app/src/main/java/tech/jpco/qen/view/MainActivity.kt

```diff
@@ -0,0 +1,91 @@
+package tech.jpco.qen.view
+
+import androidx.lifecycle.ViewModelProviders
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import android.widget.Button
+import com.jakewharton.rxbinding3.view.clicks
+import com.jakewharton.rxbinding3.view.layoutChanges
+import io.reactivex.Observable
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
+import kotlinx.android.synthetic.main.activity_main.*
+import tech.jpco.qen.R
+import tech.jpco.qen.model.SQL
+import tech.jpco.qen.viewModel.MetaEvent
+import tech.jpco.qen.viewModel.QenViewModel
+import tech.jpco.qen.viewModel.iLogger
+import java.util.concurrent.TimeUnit
+
+//Copied from https://proandroiddev.com/til-when-is-when-exhaustive-31d69f630a8b
+val <T> T.exhaustive: T
+    get() = this
+
+class MainActivity : AppCompatActivity() {
+    private val cd = CompositeDisposable()
+    private val vm by lazy { ViewModelProviders.of(this).get(QenViewModel::class.java) }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        setupInStreams()
+
+    }
+
+    override fun onStart() {
+        super.onStart()
+        setupOutStreams()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        cd.dispose()
+    }
+
+    private fun setupOutStreams() {
+        fun <T : MetaEvent> Button.throttledMetaEvent(output: T, minMillis: Long = 500): Observable<T> =
+            this.clicks()
+                .throttleFirst(minMillis, TimeUnit.MILLISECONDS)
+                .map { output }
+
+
+        val metaStream: Observable<MetaEvent> =
+            qenPage.arStream.doOnNext { iLogger("AR", it) }
+                .publish { ARs ->
```

Comment:
Starting a variable name with a capital letter makes reading the code confusing, as usually this is reserved for a class name rather than a variable name.